### ISSUE TITLE
Non-compilable code generated for nested property

### ISSUE THREAD

**Issue by agudian** [mapstruct/mapstruct#373]
We have the following declaration:

```
class Branch {
  BranchLocation branchLocation;
...
}
class BranchLocation {
   Country country;
...
}

    @Mapping(target = "countryReference", source = "branchLocation.country")
    ResultDto toResultDto(Branch entity);
```

And it generates:

```
private Country entityBranchLocationCountry(Branch branch) {
        Country country = null;
        if ( branch != null ) {
            BranchLocation branchLocation = branch.getBranchLocation();
            if ( branchLocation != null ) {
                Country country = branchLocation.getCountry();
                if ( country != null ) {
                    country = country;
                }
            }
        }
        return country;
    } 
```

So we have two declarations of `country`.

On a side note, the last null-check is superflous, as in: I wouldn't write it that way if I would create the method manually.

@sjaakd: could you take a look?

**Comment by zamrokk** [comment]
I had same problem with previous version (i.e 1.0.0 Beta3) and i confirm that the fix works in 1.0.0 RC1

**Comment by gunnarmorling** [comment]
@zamrokk Glad to hear; Thanks for your feedback :)
